mlir/utils/jenkins/Jenkinsfile

```diff
@@ -20,18 +21,50 @@ void buildProject(String target, String cmakeOpts) {
     }
 }
 
+def resetGPUs() {
+    // Abort this if runs longer than 10 minutes
+    timeout(time: 10, unit: 'MINUTES') {
+        // Run the reset, but don't fail the build if anything is wrong
+        def rc = sh(
+            script: '''
+            reset_all_gpus() {
+                echo "Scanning GPUs..."
+                GPU_IDS=$(rocm-smi | awk '/^[0-9]+[[:space:]]+[0-9]+[[:space:]]+0x/ { print $1 }')
+                if [ -z "$GPU_IDS" ]; then
+                    echo "WARNING: No GPUs found to reset."
+                    return 0
+                fi
+                for id in $GPU_IDS; do
+                    echo "Resetting GPU ID: $id"
+                    if ! rocm-smi --gpureset -d $id; then
+                        echo "WARNING: Unable to reset GPU $id"
+                    fi
+                    sleep 2
+                done
+                return 0
+            }
+            reset_all_gpus
+            ''',
+            returnStatus: true
+        )
+        if (rc != 0) {
+            echo "WARNING: reset_all_gpus exited with code ${rc}, but continuing anyway"
+        }
+    }
+}
+
 def advancedNodeCheck(Map params) {
     script {
         echo "Jenkins-side PATH = '${env.PATH}'"
     }
-    boolean doCleanWs                = params.doCleanWs
-    boolean doGPUcheck               = params.doGPUcheck
+    boolean doCleanWs = params.doCleanWs
+    boolean doGPUcheck = params.doGPUcheck
```

Comment:
Each stage is doing its own `checkNodeHealth()` which is a bit redundant given nodes are being reused across the stages. 

Would it be possible somehow to only do health checks once before any Jenkins run starts and not repeat health checks again across different stages ? 

`resetGPUs()` is taking good amount of time right now on some servers. 

Also `doCleanWs` is also true always.  does it mean it will have to clone rocMLIR each time in each stage ?